tools/cxbe/Xbe.cpp

```diff
@@ -66,7 +66,13 @@ Xbe::Xbe(const char *x_szFilename)
     {
         printf("Xbe::Xbe: Storing Xbe Path...");
 
-        strcpy(m_szPath, x_szFilename);
+        if(strlen(x_szFilename) > sizeof(m_szPath)-1)
+        {
+            SetError("Xbe Path too long", true);
+            goto cleanup;
+        }
+
+        strncpy(m_szPath, x_szFilename, sizeof(m_szPath)-1);
```

Comment:
If `strlen(x_szFilename) == sizeof(m_szPath)-1)` then this might leave `m_szPath` without a null-terminator.

Reply:
I added a new commit at the beginning which removes this variable; it was unused and also uninitialized in other constructors.
If anyone wants to reintroduce it eventually we'd probably just use `strdup` instead.